### ISSUE TITLE
Make all cron tasks exclusive per repository execution by default

### ISSUE THREAD

**Issue by carlspring** [strongbox/strongbox#487]
Note: This issue depends on #486.

We need to make it so that only one task of a given type could be running simultaneously. This should be the default behavior. There should be an option to override this, of course.

We will need a proper test case for this.

**Comment by carlspring** [comment]
@fuss86 : I assume this can be closed with your recent changes, or...?

**Comment by fuss86** [comment]
No, not all cron tasks are exclusive per repository execution by default, currently. It needs to be re-worked. See `org.carlspring.strongbox.cron.jobs.OnePerRepositoryJavaCronJob` for reference.
